pkg/util/mount/mount_linux.go

```diff
@@ -1131,18 +1172,38 @@ func doSafeOpen(pathname string, base string) (int, error) {
 	for _, seg := range segments {
 		currentPath = filepath.Join(currentPath, seg)
 		if !pathWithinBase(currentPath, base) {
-			return -1, fmt.Errorf("path %s is outside of allowed base %s", currentPath, base)
+			return -1, defaultDeviceInfo, fmt.Errorf("path %s is outside of allowed base %s", currentPath, base)
 		}
 
 		glog.V(5).Infof("Opening path %s", currentPath)
 		childFD, err = syscall.Openat(parentFD, seg, nofollowFlags, 0)
 		if err != nil {
-			return -1, fmt.Errorf("cannot open %s: %s", currentPath, err)
+			if err == syscall.ENXIO {
```

Comment:
structurally, prefer this unnested:
```
switch {
case err == syscall.ENXIO:
  ...
case err != nil:
  ...
}
```

@kubernetes/sig-storage-pr-reviews will need to weigh in on whether this handling is correct